lib/src/command/logout.dart

```diff
@@ -6,20 +6,37 @@ import 'dart:async';
 
 import '../command.dart';
 import '../oauth2.dart' as oauth2;
+import '../tokens.dart';
 
 /// Handles the `logout` pub command.
 class LogoutCommand extends PubCommand {
   @override
   String get name => 'logout';
   @override
-  String get description => 'Log out of pub.dartlang.org.';
+  String get description => 'Log out of pub.dev or any third-party pub server.';
   @override
-  String get invocation => 'pub logout';
+  String get invocation => 'pub logout [<server>] [--all]';
 
-  LogoutCommand();
+  /// Whether to log out of all servers, including third-party pub servers.
+  bool get _all => argResults['all'];
+
+  LogoutCommand() {
+    argParser.addFlag('all',
+        abbr: 'a',
+        negatable: false,
+        help: 'Log out of all servers (pub.dev and third-party pub serves).');
+  }
 
   @override
   Future run() async {
-    oauth2.logout(cache);
+    if (_all) {
+      oauth2.logout(cache);
+      removeToken(cache, all: true);
+    } else if (argResults.rest.isEmpty) {
+      oauth2.logout(cache);
+    } else {
+      final server = argResults.rest.first;
```

Comment:
Should we not validate the `server` argument? I suppose that in this case `pub logout pub.dev` or `pub logout pub.dartlang.org` could trigger `oauth2.logout(cache)`.